src/Phinx/Db/Adapter/SQLiteAdapter.php

```diff
@@ -45,6 +45,67 @@
  */
 class SQLiteAdapter extends PdoAdapter implements AdapterInterface
 {
+    // list of supported Phinx column types with their SQL equivalents
+    // some types have an affinity appended to ensure they do not receive NUMERIC affinity
+    protected static $supportedColumnTypes = [
+        self::PHINX_TYPE_BIG_INTEGER => 'biginteger',
+        self::PHINX_TYPE_BINARY => 'binary_blob',
+        self::PHINX_TYPE_BLOB => 'blob',
+        self::PHINX_TYPE_BOOLEAN => 'boolean_integer',
+        self::PHINX_TYPE_CHAR => 'char',
+        self::PHINX_TYPE_DATE => 'date_text',
+        self::PHINX_TYPE_DATETIME => 'datetime_text',
+        self::PHINX_TYPE_DOUBLE => 'double',
+        self::PHINX_TYPE_FLOAT => 'float',
+        self::PHINX_TYPE_INTEGER => 'integer',
+        self::PHINX_TYPE_JSON => 'json_text',
+        self::PHINX_TYPE_JSONB => 'jsonb_text',
+        self::PHINX_TYPE_SMALL_INTEGER => 'smallinteger',
+        self::PHINX_TYPE_STRING => 'varchar',
+        self::PHINX_TYPE_TEXT => 'text',
+        self::PHINX_TYPE_TIME => 'time_text',
+        self::PHINX_TYPE_UUID => 'uuid_text',
+        self::PHINX_TYPE_TIMESTAMP => 'timestamp_text',
+        self::PHINX_TYPE_VARBINARY => 'varbinary_blob'
+    ];
+
+    // list of aliases of supported column types
+    protected static $supportedColumnTypeAliases = [
+        'varchar' => self::PHINX_TYPE_STRING,
+        'tinyint' => self::PHINX_TYPE_SMALL_INTEGER,
+        'tinyinteger' => self::PHINX_TYPE_SMALL_INTEGER,
+        'smallint' => self::PHINX_TYPE_SMALL_INTEGER,
+        'int' => self::PHINX_TYPE_INTEGER,
+        'mediumint' => self::PHINX_TYPE_INTEGER,
+        'mediuminteger' => self::PHINX_TYPE_INTEGER,
+        'bigint' => self::PHINX_TYPE_BIG_INTEGER,
+        'tinytext' => self::PHINX_TYPE_TEXT,
+        'mediumtext' => self::PHINX_TYPE_TEXT,
+        'longtext' => self::PHINX_TYPE_TEXT,
+        'tinyblob' => self::PHINX_TYPE_BLOB,
+        'mediumblob' => self::PHINX_TYPE_BLOB,
+        'longblob' => self::PHINX_TYPE_BLOB,
+        'real' => self::PHINX_TYPE_FLOAT,
+        'numeric' => self::PHINX_TYPE_FLOAT,
```

Comment:
DECIMAL and NUMERIC are aliases in MySQL, PostgreSQL and SQL Server.

It should map to `self::PHINX_TYPE_DECIMAL`, or removed as you've removed support for DECIMAL.

Reply:
It does, which is why I had the alias. On the other hand, `NUMERIC` affinity is distinct from `FLOAT` affinity, so it would probably do more harm than good to have that alias. I'll remove it.